### DISCUSSION TITLE
setuptools.finalize_distribution_options  inplace without creating additional package like setuptools_scm

### DISCUSSION THREAD

**Discussion by hellozyemlya** [pypa/setuptools#4053] [Q&A]
Hi guys. How I can introduce hook for setuptools.finalize_distribution_options in pyproject.toml for current project? So I don't want to introduce additional package like setuptools_scm, but define some script near pyproject.toml and call it for finalize_distribution_options inplace. Is it possible to do this?

**Comment by abravalheri** [comment]
It is not possible to do that (without hacks).

If you want to customise the build for the current project, probably the best way to do that is to use `setup.py` (which you can do in <ins>***addition***</ins> to whatever you already have in `pyproject.toml` as [stated in the docs](https://setuptools.pypa.io/en/latest/references/keywords.html); the 2 are <ins>***not***</ins> mutually exclusive).
